tensorflow/core/common_runtime/zen_layout_pass.cc

```diff
@@ -0,0 +1,1165 @@
+    nb->Attr("padding", padding);
+    // If 'padding' is EXPLICIT, then 'explicit_paddings' attribute needs to be
+    // set.
+    if (padding == "EXPLICIT") {
+      std::vector<int32> explicit_paddings_tmp = {};
+      TF_CHECK_OK(GetNodeAttr(orig_node->def(), "explicit_paddings",
+                              &explicit_paddings_tmp));
+      nb->Attr("explicit_paddings", explicit_paddings_tmp);
+    }
+  }
+  nb->Attr("data_format", data_format);
+  nb->Attr("dilations", dilations);
+}
+
+// Copies the attributes from FusedConv2D op to ZenFusedConv2D op. 'padding' and
+// 'explicit_paddings' attributes are updated accordingly to PadFusedConv2D
+// fusion.
+void ZenLayoutRewritePass::UpdateZenOpAttrsFusedConv2D(const Node *orig_node,
+                                                       NodeBuilder *nb) {
+  DataType T;
+  int num_args;
+  float epsilon;
+  string data_format;
+  string padding;
+  std::vector<int32> strides;
+  std::vector<int32> dilations;
+  std::vector<int32> explicit_paddings = {};
+
+  // Get attributes from TF op node.
+  TF_CHECK_OK(GetNodeAttr(orig_node->def(), "T", &T));
+  TF_CHECK_OK(GetNodeAttr(orig_node->def(), "num_args", &num_args));
+  TF_CHECK_OK(GetNodeAttr(orig_node->def(), "strides", &strides));
+  TF_CHECK_OK(GetNodeAttr(orig_node->def(), "padding", &padding));
+  TF_CHECK_OK(GetNodeAttr(orig_node->def(), "data_format", &data_format));
+  TF_CHECK_OK(GetNodeAttr(orig_node->def(), "dilations", &dilations));
+  TF_CHECK_OK(GetNodeAttr(orig_node->def(), "epsilon", &epsilon));
+
+  // 'padding_update' determines if padding attributes needs to be modified.
+  bool padding_update = false;
+  // PadFusedConv2D fusion can be done for VALID and EXPLICIT padding.
+  if (padding != "SAME") {
+    // Check if PadFusedConv2D fusion can be done and get the padding values.
+    padding_update =
+        UpdateAttributePadConv2D(padding, orig_node, explicit_paddings);
+  }
+  // Update Zen op with attributes from TF op.
+  nb->Attr("T", T);
+  nb->Attr("num_args", num_args);
+  nb->Attr("strides", strides);
+  // Update padding attribute for PadConv2D fusion.
+  if (padding_update == true) {
+    nb->Attr("padding", "EXPLICIT");                   // Updates padding type.
+    nb->Attr("explicit_paddings", explicit_paddings);  // sets padding values.
+  } else {
+    // 'padding' attribute for condition when fusion is not performed.
+    nb->Attr("padding", padding);
+    // If 'padding' is EXPLICIT, then 'explicit_paddings' attribute needs to be
+    // set.
+    if (padding == "EXPLICIT") {
+      std::vector<int32> explicit_paddings_tmp = {};
+      TF_CHECK_OK(GetNodeAttr(orig_node->def(), "explicit_paddings",
+                              &explicit_paddings_tmp));
+      nb->Attr("explicit_paddings", explicit_paddings_tmp);
+    }
+  }
+  nb->Attr("data_format", data_format);
+  nb->Attr("dilations", dilations);
+  nb->Attr("epsilon", epsilon);
+}
+
+static void FillInputs(const Node *n,
```

Comment:
I just realized the mkl_layout_pass also has this function. But since it's a static function there shouldn't be a clash here. Since we are running out of time I'll remember to refactor this too after the branch cut. (No further actions needed for this PR.)